src/internal/errors/errors.go

```diff
@@ -33,6 +33,20 @@ var (
 	WithStack = errors.WithStack
 )
 
+// UserError is a type used to indicate that an error is not an internal error
+// so that we can log it differently (i.e. do not log at the ERROR level with a
+// stack trace)
+type UserError struct{}
```

Comment:
There is another package `pacherr` which has some of the structured error stuff.  I have been using that for structured error handling, while trying to phase out the string matching error handling in this package.

Separately: What do you think about having a `Hint` field here, or something that allows us to communicate helpful information to the user about how to resolve the error?

Reply:
I don't think this additional type is the best approach, see my comment below